### ISSUE TITLE
generate valid context.jsonld from vocab

### ISSUE THREAD

**Issue by ahdinosaur** [valueflows/agent#19]
from our [vocab](https://github.com/openvocab/agent/#agent-vocab) we should generate a valid '@context' for our RDF / JSON-LD objects, but i'm not sure what the syntax is.

TODO:
- [ ] generate valid context.jsonld

references:
- [simplerdf#context](https://github.com/nicola/simplerdf#tip)

@elf-pavlik do you have any references to the context format in more detail? as in, what do `@id` and `@type` mean in a context?

**Comment by fosterlynn** [comment]
@ahdinosaur Yes! Thanks!

I have learned some things about the syntax, I can help some.  Although still way behind @elf-pavlik !  (So please correct if I am wrong below.)  And I also have some questions around context in my generated json-ld, still studying.

@id is the URI reference for the "object".
@type is the class (rdfs:Class) - like could be a rdf:Property or a foaf:Person, etc.

If you like, I'll just plug in and help on the context generation.  And of course, we need to make some decisions on the vocab itself and existing vocabs to use.

**Comment by fosterlynn** [comment]
@ahdinosaur I'm going to start working on the context if that is OK with you.  Let me know if you would rather do it!  Or if we could work together in some way.  Although I think given the task and the time differences, that would be kind of hard.

**Comment by ahdinosaur** [comment]
@fosterlynn that sounds great, super keen for us to get a proper `@context`. :) then later when i have capacity i'll use it as an example to fix my [schema-jsonld-context](https://www.npmjs.com/package/schema-jsonld-context) module.

**Comment by elf-pavlik** [comment]
- http://www.w3.org/TR/json-ld/#the-context

`@type` inside of `@context` works bit differently than in the rest of the document! You may need to understand this construct:

**`"foo": { "@type": "@id" }`** - defines foo as [ObjectProperty (not DataProperty)](http://www.w3.org/TR/2012/REC-owl2-primer-20121211/#Entity_Declarations) and will assume its values being an URIs or nested objects with `@id` (or without it - Blank Node)

AFAIK if you use **`"foo": { "@type": "ex:blah" }`** it will define foo as DataProperty which values - Literals (strings) not URIs or nested objects! have **datatype** _ex:blah_

**Comment by fosterlynn** [comment]
@elf-pavlik thanks, studying.

**Comment by fosterlynn** [comment]
@ahdinosaur @elf-pavlik I created a very rough initial draft of @context for agent data.  I decided that it would be the start to one @context doc for the whole vocab, so I created a new repo for this and a webpage for the doc, and whatever else we need for the vocab officially.

I have questions and notes.  (I could probably take it further with some more research, but wanted to get it out there and socialize it a bit.)  Here are some:
1. I added a class Network to the Agent class hierarchy, thought it might be nice to have that explicitly defined.  I'm not that tied to it, could be there or not.  Made it a subclass of Group, not sure if people would agree. 
2. Need to figure out how to get the relationship type stuff in there:  subject(type), object(type), context(type), label, inverse label.
3. Note there will be more classes in the domain of "context" later.
4. Can we specify an existing property type like name: foaf:name, and also specify other things like a domain that is not specified in foaf:name?  I think so, but....
5. Same kind of question as 4. I used as:Relationship, but also added more properties.  OK to do that?

You will probably have questions for me; and probably advice on how to do it better!!  Which is very welcome.
